crates/gitbutler-branch-actions/src/squash.rs

```diff
@@ -192,6 +192,13 @@ fn validate(
     source_commits: &[git2::Commit<'_>],
     destination_commit: &git2::Commit<'_>,
 ) -> Result<()> {
+    if source_commits
```

Comment:
If there are multiple commits A, B, C, could I not say to squash it all into `B`? This seems not unreasonable.

Reply:
Why should B be a source AND the destination?

Reply:
Should you not say you're squashing A and C into B?

Reply:
Ah, right, so that source commits never contain the destination. Yes, that's fair.
Since the user is making the selections though, maybe we could auto-correct what's correctable? Either in the frontend or in the backend?